File: dashboard/sim_paths.py
"""Canonical allowlist of simulator-relevant repo paths.

Dirty-run captures (``dashboard/state_capture.py``) and the dashboard UI
(``dashboard/web/lib/diff.ts`` via ``dashboard/web/lib/sim-paths.ts``) both
scope their diffs to files that can actually affect simulator behaviour.
Changes to dashboard code, scratch scripts, docs, or local test files cannot
move a testcase result; showing them in the diff just adds noise and erodes
trust in the dashboard's core board-facing question 3 ("what changed in
simulator code/config between two runs?").

Mirror any change to this list in ``dashboard/web/lib/sim-paths.ts``.
"""

from __future__ import annotations

from pathlib import PurePosixPath

SIMULATOR_PATH_PREFIXES: tuple[str, ...] = (
    "Base_classes/",
    "assets/",
    "skills/",
    "testcases/",
    "fighters_data/",
    "battle_specs_manual/",
)

SIMULATOR_ROOT_FILES: frozenset[str] = frozenset(
    {
        "pyproject.toml",
        "check_testcases.py",
        "battle_main.py",
        "compare_results.py",
    }
)


def is_simulator_path(rel_path: str) -> bool:
    """Return True if ``rel_path`` (repo-relative, POSIX-style) is simulator-relevant."""
    if not rel_path:
        return False
    normalized = str(PurePosixPath(rel_path))
    if normalized in SIMULATOR_ROOT_FILES:
        return True
    return any(normalized.startswith(prefix) for prefix in SIMULATOR_PATH_PREFIXES)


def git_pathspec_args() -> tuple[str, ...]:
    """Pathspec list suitable for ``git diff -- <pathspec>...`` calls."""
    return tuple(SIMULATOR_PATH_PREFIXES) + tuple(sorted(SIMULATOR_ROOT_FILES))

File: dashboard/web/lib/sim-paths.ts
// Canonical allowlist of simulator-relevant repo paths.
// Mirror of dashboard/sim_paths.py — keep the two lists in lockstep.
// Changes to dashboard code, scratch scripts, docs, or local test files cannot
// move a testcase result, so diffs scope to this list to answer board question
// 3 cleanly ("what changed in simulator code/config between two runs?").

// Paths reflect the monorepo layout: the TypeScript simulator engine and native
// config live under simulator/. Fighter stat profiles shared with scripts live
// under shared/. The ground-truth testcase corpus stays at the repo root because
// its path is a stable logical id.
export const SIMULATOR_PATH_PREFIXES: readonly string[] = [
  "simulator/config/",
  "simulator/src/",
  "shared/fighters_data/",
  "testcases/",
];

export const SIMULATOR_ROOT_FILES: ReadonlySet<string> = new Set<string>([
  "simulator/package.json",
  "simulator/package-lock.json",
  "simulator/tsconfig.json",
]);

export function isSimulatorPath(relPath: string | undefined | null): boolean {
  if (!relPath) return false;
  const p = relPath.replace(/^[ab]\//, "");
  if (!p) return false;
  if (SIMULATOR_ROOT_FILES.has(p)) return true;
  for (const prefix of SIMULATOR_PATH_PREFIXES) {
    if (p.startsWith(prefix)) return true;
  }
  return false;
}
